src/utils/topic.ts

```diff
@@ -2,117 +2,111 @@ import { ref, watch, type Ref } from 'vue'
-          .filter((id) => id != null)
-      )
-    ]
-  }
+  const factors = (
+    await useTopicElementStore().getTopicElements<Factor>({ topicId: topic.id })
+  ).map((element) => {
+    if (!keepDatasets) {
+      element.element = null
+      element.extras[topicsExtrasKey].uri = null
+      element.extras[topicsExtrasKey].availability = Availability.NOT_AVAILABLE
+    }
+    return element
+  })
 
   return {
     ...data,
     private: true,
-    datasets: keepDatasets ? getDatasetsIds() : [],
-    reuses: [],
     spatial: undefined,
     owner: useUserStore().data ?? null,
     organization: null,
-    extras: updateTopicExtras(topic, {
-      cloned_from: topic.id,
-      datasets_properties: topic.extras[
-        topicsExtrasKey
-      ].datasets_properties.map((dp) => {
-        return {
-          ...dp,
-          id: keepDatasets ? dp.id : null,
-          uri: keepDatasets ? dp.uri : null,
-          availability: keepDatasets
-            ? dp.availability
-            : Availability.NOT_AVAILABLE
-        }
-      })
-    })
+    elements: factors,
+    // we're not copying all the extras over, only the ones we control
+    extras: {
+      [topicsExtrasKey]: {
+        cloned_from: topic.id
+      }
+    }
   }
 }
 
 export function useExtras(topic: Ref<Topic | null | undefined>): {
-  datasetsProperties: Ref<DatasetProperties[]>
   clonedFrom: Ref<Topic | null>
 } {
-  const datasetsProperties: Ref<DatasetProperties[]> = ref([])
   const clonedFrom = ref<Topic | null>(null)
 
   watch(
     topic,
     () => {
       const extras = topic.value?.extras[topicsExtrasKey]
-      if (extras != null) {
-        datasetsProperties.value = extras.datasets_properties ?? []
+      if (extras?.cloned_from != null) {
+        useTopicStore()
+          .load(extras.cloned_from, { toasted: false })
+          .then((res) => {
+            clonedFrom.value = res
+          })
+          .catch((err) => {
+            console.error('Failed fetching cloned_from', err.response?.data)
+            clonedFrom.value = null
+          })
+      }
+    },
+    { immediate: true }
+  )
+
+  return { clonedFrom }
+}
+
+export function useTopicFactors(topic: Ref<Topic | null | undefined>): {
+  factors: Ref<ResolvedFactor[]>
+  nbFactors: Ref<number>
+} {
+  const nbFactors: Ref<number> = ref(0)
+  const factors: Ref<ResolvedFactor[]> = ref([])
 
-        if (extras.cloned_from != null) {
-          useTopicStore()
-            .load(extras.cloned_from, { toasted: false })
-            .then((res) => {
-              clonedFrom.value = res
-            })
-            .catch((err) => {
-              console.error('Failed fetching cloned_from', err.response?.data)
-              clonedFrom.value = null
-            })
-        } else {
-          clonedFrom.value = null
+  watch(
+    topic,
+    async () => {
+      if (!topic.value) return
+      const rawElements = await useTopicElementStore().getTopicElements<Factor>(
```

Comment:
Same s/// possible here and in the .map below, although I'm not sure what getTopicElements<Factor> actually types at runtime? A GenericElement or a Factor?